server/endpoint/status/endpoint.go

```diff
@@ -0,0 +1,158 @@
+	Method = "PATCH"
+	// Name identifies the endpoint. It is aligned to the package path.
+	Name = "status/updater"
+	// Path is the HTTP request path this endpoint is registered for.
+	Path = "/status/{app_namespace}/{app_name}/"
+)
+
+type Config struct {
+	K8sClient k8sclient.Interface
+	Logger    micrologger.Logger
+}
+
+type Endpoint struct {
+	k8sClient k8sclient.Interface
+	logger    micrologger.Logger
+}
+
+func New(config Config) (*Endpoint, error) {
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+	if config.K8sClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.K8sClient must not be empty", config)
+	}
+
+	e := &Endpoint{
+		logger:    config.Logger,
+		k8sClient: config.K8sClient,
+	}
+
+	return e, nil
+}
+
+func (e Endpoint) Decoder() kithttp.DecodeRequestFunc {
+	return func(ctx context.Context, r *http.Request) (interface{}, error) {
+		var request Request
+
+		defer r.Body.Close()
+		err := json.NewDecoder(r.Body).Decode(&request)
+		if err != nil {
+			return nil, microerror.Maskf(decodeFailedError, "%v", err.Error())
+		}
+
+		namespace, _ := ctx.Value("app_namespace").(string)
+		name, _ := ctx.Value("app_name").(string)
+		request.AppNamespace = namespace
+		request.AppName = name
+
+		return request, nil
+	}
+}
+
+func (e Endpoint) Encoder() kithttp.EncodeResponseFunc {
+	return func(ctx context.Context, w http.ResponseWriter, response interface{}) error {
+		w.Header().Set("Content-Type", "application/json; charset=utf-8")
+
+		return json.NewEncoder(w).Encode(response)
+	}
+}
+
+func (e Endpoint) Endpoint() kitendpoint.Endpoint {
+	return func(ctx context.Context, r interface{}) (interface{}, error) {
+		var err error
+
+		request := r.(Request)
+
+		desiredStatus := v1alpha1.AppStatus{
+			AppVersion: request.AppVersion,
+			Release: v1alpha1.AppStatusRelease{
+				LastDeployed: request.LastDeployed,
+				Reason:       request.Reason,
+				Status:       request.Status,
+			},
+			Version: request.Version,
+		}
+
+		// Get app CR again to ensure the resource version is correct.
+		currentCR, err := e.k8sClient.G8sClient().ApplicationV1alpha1().Apps(request.AppNamespace).Get(ctx, request.AppName, metav1.GetOptions{})
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+
+		if equals(currentCR.Status, desiredStatus) {
+			// no-op
+			return nil, nil
+		}
+
+		e.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("setting status for app %#q in namespace %#q", request.AppName, request.AppNamespace))
+
+		currentCR.Status = desiredStatus
+
+		_, err = e.k8sClient.G8sClient().ApplicationV1alpha1().Apps(request.AppNamespace).UpdateStatus(ctx, currentCR, metav1.UpdateOptions{})
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+
+		e.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("status set for app %#q in namespace %#q", request.AppName, request.AppNamespace))
+
+		return nil, nil
```

Comment:
Return nothing but 200 response in case of no error, is that okay?

Reply:
Yes 200 makes sense since it's updating the status. I thought maybe 202 Accepted but the processing is not async.